packages/react/src/components/Popup/Popup.tsx

```diff
@@ -120,6 +133,35 @@ function useRestoreFocus(props: IPopupProps, root: React.RefObject<HTMLDivElemen
   );
 }
 
+function ariaHidden(nodesToHide: HTMLElement[], show: boolean) {
+  if (show) {
+    nodesToHide.forEach(child => child.setAttribute('aria-hidden', 'true'));
+  } else {
+    nodesToHide.forEach(child => child.removeAttribute('aria-hidden'));
+  }
+}
+
+function useGetNodesToHide(props: IPopupProps): HTMLElement[] | undefined {
+  const [nodesToHide, setNodesToHide] = React.useState<HTMLElement[]>();
+  const isModalOrPanel = props['aria-modal'];
+
+  React.useEffect(() => {
+    if (isModalOrPanel) {
+      const blackListTagNames = ['TEMPLATE', 'SCRIPT', 'STYLE'];
+      const bodyChildren = getChildren(getDocument()!.body);
+
+      const filteredBodyChildren = bodyChildren
+        .slice(0, bodyChildren.length - 1)
+        .filter(child => !blackListTagNames.includes(child.tagName) && !child.hasAttribute('aria-hidden'));
+
+      setNodesToHide(filteredBodyChildren);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps -- should only run on first render
```

Comment:
Is this assumption true? What if the underlying body changes while the `Popup` is still active?

Reply:
Good point - i'll remove in conjunction with the changes Ben suggested